Microsoft.HealthVault.Fhir/Transformers/SleepJournalAmToFhir.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// MIT License
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the ""Software""), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+using System;
+using System.Collections.Generic;
+using Hl7.Fhir.Model;
+using Microsoft.HealthVault.Fhir.Codings;
+using Microsoft.HealthVault.Fhir.Constants;
+using Microsoft.HealthVault.ItemTypes;
+
+namespace Microsoft.HealthVault.Fhir.Transformers
+{
+    public static partial class ThingBaseToFhir
+    {
+        // Register the type on the generic ThingToFhir partial class
+        public static Observation ToFhir(this SleepJournalAM sleepJournalAM)
+        {
+            return SleepJournalAMToFhir.ToFhirInternal(sleepJournalAM, ThingBaseToFhir.ToFhirInternal(sleepJournalAM));
+        }
+    }
+
+    /// <summary>
+    /// An extension class that transforms HealthVault sleep journal am data types into FHIR Observations
+    /// </summary>
+    internal static class SleepJournalAMToFhir
+    {
+        internal static Observation ToFhirInternal(SleepJournalAM sleepJournalAM, Observation observation)
+        {
+            observation.Category = new List<CodeableConcept>() { FhirCategories.VitalSigns };
```

Comment:
@gcorvera - It looks like the preferred code system to use is: https://www.hl7.org/fhir/codesystem-observation-category.html. 

Out of those, "survey" seems to be the closest match for sleep and exercise. Do you think we should go with survey or go with our own?